src/sentry/tasks/files.py

```diff
@@ -36,5 +37,7 @@ def delete_unreferenced_blobs(blob_ids):
         try:
             # Need to delete the record to ensure django hooks run.
             FileBlob.objects.get(id=blob_id).delete()
-        except FileBlob.DoesNotExist:
+        except (IntegrityError, FileBlob.DoesNotExist):
```

Comment:
You need to wrap this in an atomic block then, otherwise it’ll taint the transaction.

Reply:
You might wanna move other database operations out of this transaction now just to limit it, but probably not a big deal relatively.

Reply:
```python
try:
  fb = FileBlob.objects.get(id=file_blob_id)
except FileBlob.DoesNotExist:
  pass
else:
  try:
    with transaction.atomic():
      fb.delete()
  except IntegrityError:
    pass
```

Would be the slightly better pattern here.